### PR TITLE
NCTL :: enhancements, standardisation & regression fix

### PR DESCRIPTION
**Pull request by ghost**
1. Standardised bash style.
2. Docs updated.
3. New command to view main purse uref.
4. Further erc-20 changes.
5. Folder structure simplification.

### CODE REVIEW

File: utils/nctl/sh/assets/setup.sh [hunk 1]
```diff
@@ -63,14 +63,14 @@ function _set_chainspec() {
...
+    # Override config settings as all paths need to point relative to nctl's assets dir:
+    #    genesis.accounts_path
     #    genesis.mint_installer_path
     #    genesis.pos_installer_path
     #    genesis.standard_payment_installer_path
     #    genesis.auction_installer_path
-    # These are paths to WASM smart contracts, and they still need to point
-    # relative to nctl's assets dir.
-    sed -i "s?\.\./\.\./target/wasm32-unknown-unknown/release/?../bin/?g" $path_config > /dev/null 2>&1
+    sed -i "s?\${BASEDIR}/target/wasm32-unknown-unknown/release/?../bin/?g" $path_config > /dev/null 2>&1
```
Comment (linclelinkpart5) [inline utils/nctl/sh/assets/setup.sh:72]: This might need to have the backslash before `${BASEDIR}` removed?

**Review by linclelinkpart5 — APPROVED** [review]
Just the backslash before `${BASEDIR}` might be off, but I'm not sure off the top of my head. Otherwise, looks good to me!

File: utils/nctl/sh/assets/setup.sh (hunk 1 above)
Comment (momipsl) [reply utils/nctl/sh/assets/setup.sh:72]: No is good ... tested it a few times.

**Comment by linclelinkpart5** [comment]
Drone failed on this PR, merging in latest master and trying again.

**Comment by linclelinkpart5** [comment]
bors try

**Pull request merged**